### ISSUE TITLE
Feature, relation chain

### ISSUE THREAD

**Issue by grzegorz-pierzakowski** [phalcon/cphalcon#12552]
Expected:
I whsh to define a relation with multiple joins as below:
```php
$this->hasMany('id', Car::class, 'my_class_id', [
'with' => 'tires'
]);
```
where tires i a onToMany ralation alias on Car.

For now it's not working that way.

I am trying to get related modal data with one sql query and want the object + appropriate relations to be build then.

**Comment by Jurigag** [comment]
? Just use query builder:

```php
$result = $modelsManager->createBuilder()
->columns('Salon.*, Car.*, Tires.*)
->from('Salon')
->leftJoin('Car')
->leftJoin('Tires')
->getQuery()
->execute()
```

Eager loading is bad and won't be implemented in phalcon. check https://github.com/phalcon/cphalcon/issues/1117 especially @andresgutierrez comments

**Comment by grigoriygithub** [comment]
@Jurigag.  Eager loading very helps in the some cases. In your solution, you cannot get a model and its relationships like that:
- Model
  - RelatedModel

**Comment by Jurigag** [comment]
Yes but read @andresgutierrez comments, eager loading is bad for performance and caching.

**Comment by grigoriygithub** [comment]
But question of eager loading is asked regularly. Maybe, the best solutions is implementation but with notification to developers about performance issues.

**Comment by Jurigag** [comment]
This is why eager loading is already in incubator.

**Comment by grigoriygithub** [comment]
Eager loading in incubator have two issues:
1. It does not work correctly with getters/setters in model;
2. Eager Result not saved as reusable records in Models Manager

Prevents to resolve these issues: https://github.com/phalcon/cphalcon/blob/master/phalcon/mvc/model/manager.zep#L1352
How to emulate unique_key in PHP?

**Comment by sergeyklay** [comment]
Feel free to send PR with fixes for 1 and 2
